test/upsert/mzcompose.py

```diff
@@ -178,3 +179,72 @@ def workflow_rehydration(c: Composition) -> None:
 
         c.run("testdrive", "rehydration/04-reset.td")
         c.kill("clusterd1")
+
+
+def workflow_failpoint(c: Composition) -> None:
+    """Test behaviour when upsert state errors"""
+    print("Running failpoint workflow")
+
+    c.down(destroy_volumes=True)
+    c.up("materialized")
+    c.run("testdrive", "failpoint/01-setup.td")
+
+    for failpoint in [
+        (
+            "fail_merge_snapshot_chunk",
+            "failure while rehydrating state: Error merging snapshot values",
+        ),
+        (
+            "fail_state_multi_put",
+            "failure while updating records in state: Error putting values into state",
+        ),
+        (
+            "fail_state_multi_get",
+            "failure while fetching records from state: Error getting values from state",
+        ),
+    ]:
+        run_one_failpoint(c, failpoint[0], failpoint[1])
```

Comment:
this is so awesome!

Reply:
\o/

Reply:
(To give it's due, it's a ~copy pasta from one of persist's failpoint test)